homeassistant/components/miele/button.py

```diff
@@ -0,0 +1,166 @@
+            MieleAppliance.TUMBLE_DRYER,
+            MieleAppliance.TUMBLE_DRYER_SEMI_PROFESSIONAL,
+            MieleAppliance.DISHWASHER,
+            MieleAppliance.OVEN,
+            MieleAppliance.OVEN_MICROWAVE,
+            MieleAppliance.STEAM_OVEN,
+            MieleAppliance.MICROWAVE,
+            MieleAppliance.HOOD,
+            MieleAppliance.WASHER_DRYER,
+            MieleAppliance.STEAM_OVEN_COMBI,
+            MieleAppliance.STEAM_OVEN_MICRO,
+            MieleAppliance.STEAM_OVEN_MK2,
+            MieleAppliance.DIALOG_OVEN,
+        ),
+        description=MieleButtonDescription(
+            key="stop",
+            translation_key="stop",
+            press_data={PROCESS_ACTION: MieleActions.STOP},
+        ),
+    ),
+    MieleButtonDefinition(
+        types=(
+            MieleAppliance.WASHING_MACHINE,
+            MieleAppliance.WASHING_MACHINE_SEMI_PROFESSIONAL,
+            MieleAppliance.TUMBLE_DRYER,
+            MieleAppliance.TUMBLE_DRYER_SEMI_PROFESSIONAL,
+            MieleAppliance.DISHWASHER,
+            MieleAppliance.WASHER_DRYER,
+        ),
+        description=MieleButtonDescription(
+            key="pause",
+            translation_key="pause",
+            press_data={PROCESS_ACTION: MieleActions.PAUSE},
+        ),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: MieleConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the button platform."""
+    coordinator = config_entry.runtime_data
+
+    async_add_entities(
+        MieleButton(coordinator, device_id, definition.description)
+        for device_id, device in coordinator.data.devices.items()
+        for definition in BUTTON_TYPES
+        if device.device_type in definition.types
+    )
+
+
+class MieleButton(MieleEntity, ButtonEntity):
+    """Representation of a Button."""
+
+    entity_description: MieleButtonDescription
+
+    def __init__(
+        self,
+        coordinator: MieleDataUpdateCoordinator,
+        device_id: str,
+        description: MieleButtonDescription,
+    ) -> None:
+        """Initialize the button."""
+        super().__init__(coordinator, device_id, description)
+        self.api = coordinator.api
+
+    def _action_available(self, action: dict[str, str | int | bool]) -> bool:
+        """Check if action is available according to API."""
+        return (
+            action[PROCESS_ACTION]
+            in self.coordinator.data.actions[self._device_id].process_actions
+        )
+
+    @property
+    def available(self) -> bool:
+        """Return the availability of the entity."""
+
+        return super().available and self._action_available(
+            self.entity_description.press_data
+        )
+
+    async def async_press(self) -> None:
+        """Press the button."""
+        _LOGGER.debug("Press: %s", self.entity_description.key)
+        if self._action_available(self.entity_description.press_data):
```

Comment:
Why the if statement?

Reply:
It is guarding that the action is not sent if button is unavailable. But HA perhaps takes care of that?

Reply:
It should!